ui/src/app/utils/access-utils.tsx

```diff
@@ -389,7 +390,7 @@ export const useIsUserDisabled = () => {
   return disabled;
 };
 
-export const useNeedsToAcceptTOS = () => {
+export const useShowTOS = () => {
   const { authLoaded, isSignedIn } = useStore(authStore);
   const profile = profileStore.get().profile;
   const [userRequiredToAcceptTOS, setUserRequiredToAcceptTOS] =
```

Comment:
This does not look like something that you added, but do you know why this is useState? I am not familiar with a useState inside a useEffect.

Reply:
It looks like it might be a pattern in custom hooks. Interesting:
https://reactjs.org/docs/hooks-custom.html

Reply:
I thought that calling an async function from within a function would necessitate that the containing function is asynchronous as well. It seems like that is not the case with useEffects.

https://www.designcise.com/web/tutorial/how-to-use-async-await-in-react-useeffect-hook#create-a-self-invoking-anonymous-function